### PR TITLE
Added a shutdown strategy for db-scheduler 

### PR DESCRIPTION
**Pull request by MatthewStocks**
## Context

Adding a strategy to handle graceful shutdown for apps that enable the db-scheduler bean 

## Checklist
- [x] Change meets or does not compromise the [Baseline Security Requirements](https://transferwise.atlassian.net/wiki/spaces/EKB/pages/434929973/Baseline+Security+Requirements)

### CODE REVIEW

File: core/build.gradle [hunk 1]
```diff
@@ -14,6 +14,8 @@ dependencies {
     implementation libraries.springBootActuator
     implementation libraries.springWeb
     implementation libraries.guava
+    implementation libraries.dbSchedulerSpringBootStarter
```
Comment (simondwilliams) [inline core/build.gradle:17]: Can we make this a compile-time dependency and use `@ConditonalOnClass`?

Comment (MatthewStocks) [reply core/build.gradle]: changed in latest

File: core/src/main/java/com/transferwise/common/gracefulshutdown/GracefulShutdownAutoConfiguration.java [hunk 2]
```diff
@@ -1,7 +1,9 @@
 package com.transferwise.common.gracefulshutdown;
 
+import com.github.kagkarlsson.scheduler.Scheduler;
```
Comment (simondwilliams) [inline core/src/main/java/com/transferwise/common/gracefulshutdown/GracefulShutdownAutoConfiguration.java:3]: These look like they can be reverted now

**Review by simondwilliams — APPROVED** [review]

**Review by onukristo — CHANGES_REQUESTED** [review]
@Component should never be used in autoconfigured libraries.

No test coverage :).

Strategy class in wrong package.

scheduler.stop() is blocking method. We have to calle it from another thread.

**Pull request closed without merging**